gym/envs/mujoco/mujoco_env.py

```diff
@@ -9,41 +9,49 @@
 
 try:
     import mujoco_py
-    from mujoco_py.mjlib import mjlib
+    from mujoco_py import load_model_from_path, MjSim, MjViewer
 except ImportError as e:
     raise error.DependencyNotInstalled("{}. (HINT: you need to install mujoco_py, and also perform the setup instructions here: https://github.com/openai/mujoco-py/.)".format(e))
 
 class MujocoEnv(gym.Env):
     """Superclass for all MuJoCo environments.
     """
 
-    def __init__(self, model_path, frame_skip):
+    def __init__(self, model_path, frame_skip, resolution=(240, 240), render=False):
         if model_path.startswith("/"):
             fullpath = model_path
         else:
             fullpath = os.path.join(os.path.dirname(__file__), "assets", model_path)
         if not path.exists(fullpath):
             raise IOError("File %s does not exist" % fullpath)
+        self.resolution = resolution
         self.frame_skip = frame_skip
-        self.model = mujoco_py.MjModel(fullpath)
-        self.data = self.model.data
-        self.viewer = None
+        self.model = load_model_from_path(fullpath)
+        self.sim = MjSim(self.model)
+        self.data = self.sim.data
+        if render:
+            self.viewer = MjViewer(self.sim)
+        else:
+            self.viewer = None
 
         self.metadata = {
             'render.modes': ['human', 'rgb_array'],
             'video.frames_per_second': int(np.round(1.0 / self.dt))
         }
 
-        self.init_qpos = self.model.data.qpos.ravel().copy()
-        self.init_qvel = self.model.data.qvel.ravel().copy()
+        self.init_qpos = self.data.qpos.ravel().copy()
+        self.init_qvel = self.data.qvel.ravel().copy()
         observation, _reward, done, _info = self._step(np.zeros(self.model.nu))
         assert not done
         self.obs_dim = observation.size
 
-        bounds = self.model.actuator_ctrlrange.copy()
-        low = bounds[:, 0]
-        high = bounds[:, 1]
-        self.action_space = spaces.Box(low, high)
+        bounds = self.model.actuator_ctrlrange
+        # I'm not sure why bounds is at least sometimes None ... bug?
```

Comment:
@adamlerer / @yayitsamyzhang could you share examples of when this happens?  Possibly a bug.

Reply:
Unfortunately I've lost track of when this happened. You could try setting it to None and then run an environment and see what happens!

Reply:
It's constructed by generated code (here in the most recent version: https://github.com/openai/mujoco-py/blob/master/mujoco_py/generated/wrappers.pxi#L1285).

Wrapper is defined here: https://github.com/openai/mujoco-py/blob/master/mujoco_py/generated/wrappers.pxi#L3824

So it looks like we set it to `None` instead of a numpy array with 0 as one of the dimensions (which would just be empty anyways).

This would be true for models without any actuators (so all the `ctrl` related arrays are size 0).

I'm alright with this handling -- maybe update the comment to say that it's None when there are no actuators.

If you wanted to, you could check `sim.model.nu == 0`, which is the same thing (no actuators in the model).